packages/babel-parser/src/options.ts

```diff
@@ -100,13 +100,13 @@ export function getOptions(opts?: Options | null): OptionsWithDefaults {
     throw new Error("The `annexB` option can only be set to `false`.");
   }
 
-  const options: any = {};
+  // https://github.com/babel/babel/pull/16918
+  // `options` is accessed frequently, please make sure it is a fast object.
+  // `%ToFastProperties` can make it a fast object, but the performance is the same as the slow object.
+  const options: any = { ...defaultOptions };
```

Comment:
Nit: If we move the assignment to the start, the `opts == null` branch can just reuse the `options`.